src/web-auth/captcha.js

```diff
@@ -0,0 +1,117 @@
+// eslint-disable-next-line no-unused-vars
+import Authentication from '../authentication';
+import object from '../helper/object';
+
+var defaults = {
+  lang: 'en',
+  templates: {
+    'auth0': function (challenge) {
+      var message = challenge.type === 'code' ?
+        'Enter the code shown above' :
+        'Solve the formula shown above';
+      return '<div class="captcha-challenge">\n' +
+        '  <img src="' + challenge.image + '" />\n' +
+        '  <button type="button" class="captcha-reload">↺</button>\n' +
+        '</div>\n' +
+        '<input type="text" name="captcha"\n' +
+        '  class="form-control captcha-control"\n' +
+        '  placeholder="' + message + '" />';
+    }
+    ,
+    'recaptcha_v2': function () {
+      return '<div class= "recaptcha" ></div><input type="hidden" name="captcha" />';
+    }
+  }
+};
+
+function handleAuth0Provider(element, options, challenge, load) {
+  element.innerHTML = options.templates[challenge.provider](challenge);
+  element.querySelector('.captcha-reload').addEventListener('click', function (e) {
+    e.preventDefault();
+    load();
+  });
+}
+
+function injectRecaptchaScript(element, lang, callback) {
+  var callbackName = 'recatpchaCallback_' + Math.floor(Math.random() * 1000001);
```

Comment:
@jfromaniello typo🧐🧐🧐 this should be caPtcha (it's tricky to say it even for me)

Reply:
you are right